v3/src/lib/analytics.ts

```diff
@@ -0,0 +1,78 @@
+enum AnalyticsCategory {
+  GENERAL = "general",  // catch-all for events that we haven't categorized yet
+  DATA = "data",     // Interactions with generated data
+  DOCUMENT = "document", // Document interactions
+  PLUGIN = "plugin",     // Game interactions
+  MODEL = "model",    // Model interactions
+  PLOT = "plot",     // Plot interactions
+  SESSION = "session",  // Session events (log in/out)
+  COMPONENT = "component", // General component interactions
+  TABLE = "table",    // Table interactions
+  SLIDER = "slider",    // Slider interactions
+  TEXT = "text",     // Text interactions
+}
```

Comment:
One other note on this -- you don't need named constants here because TypeScript types can guard against typos. The way we've handled this elsewhere is:
```typescript
export const AnalyticsCategories = [
  "general",      // catch-all for events that we haven't categorized yet
  "data",         // Interactions with generated data
  ...
  "text"          // Text interactions
] as const
export type AnalyticsCategory = typeof AnalyticsCategories[number]
```